agents/vantagePro2_agent/vantage_pro2_agent.py

```diff
@@ -0,0 +1,213 @@
+        self.lock = TimeoutLock()
+
+        self.port = port
+        self.module: Optional[VantagePro2] = None
+
+        if sample_freq > 0.5:
+            self.log.warn("Sample frequency too fast! Setting to 0.5Hz")
+            sample_freq = 0.5
+        self.sample_freq = sample_freq
+
+        self.initialized = False
+        self.take_data = False
+
+        # Registers weather data feed
+        agg_params = {
+            'frame_length': 60,
+        }
+        self.agent.register_feed('weather_data',
+                                 record=True,
+                                 agg_params=agg_params)
+
+    # Task functions.
+    def init_VantagePro2_task(self, session, params=None):
+        """
+        Perform first time setup of the Weather Monitor Module.
+
+        Args:
+            params (dict): Parameters dictionary for passing parameters to
+                task.
+
+        """
+        if params is None:
+            params = {}
+
+        auto_acquire = params.get('auto_acquire', False)
+
+        if self.initialized:
+            return True, "Already Initialized Module"
+
+        with self.lock.acquire_timeout(0, job='init') as acquired:
+            if not acquired:
+                self.log.warn("Could not start init because "
+                              "{} is already running".format(self.lock.job))
+                return False, "Could not acquire lock."
+
+            session.set_status('starting')
+
+            self.module = VantagePro2(self.port)
+            print("Initialized Vantage Pro2 module: {!s}".format(
+                self.module))
+
+        self.initialized = True
+
+        # Start data acquisition if requested
+        if auto_acquire:
+            self.agent.start('acq')
+
+        return True, 'Vantage Pro2 module initialized.'
+
+    def start_acq(self, session, params=None):
+        """
+        Method to start data acquisition process.
+
+        Args:
+            sample_freq (double):
+                Frequency at which weather data is sampled.
+                Defaults to 0.5 Hz.
+
+        """
+        time.sleep(2)
+        if params is None:
+            params = {}
+
+        sample_freq = params.get('sample_freq')
+        # If loops is None, use value passed to Agent init
+        if sample_freq is None:
+            sample_freq = self.sample_freq
+        wait_time = 1/sample_freq
+
+        with self.lock.acquire_timeout(0, job='acq') as acquired:
+            if not acquired:
+                self.log.warn("""Could not start acq because {} is
+                already running"""
+                              .format(self.lock.job))
+                return False, "Could not acquire lock."
+
+            session.set_status('running')
+            # use pacemaker class to take data at regular intervals
+            if sample_freq % 1 == 0:
+                pm = Pacemaker(sample_freq, True)
+            else:
+                pm = Pacemaker(sample_freq)
```

Comment:
Oops looking back now this is definitely a problem with the Pacemaker class and quantize should also work with sample frequencies that are 1/n Hz... my bad